webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -278,4 +304,145 @@ describe('TasksComponent', () => {
+          { name: 'Chattanooga Village' },
+          null,
+          null,
+        ],
+      },
+    ];
+
+    it('should not alter tasks lineage if user is online only', async () => {
+      const expectedTasks = [
+        {
+          _id: 'e1',
+          date: moment('2020-10-20').toDate(),
+          dueDate: '2020-10-20',
+          lineage: [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area' ],
+          overdue: true,
+          owner: 'a',
+          forId: 'a',
+        },
+        {
+          _id: 'e2',
+          date: moment('2020-10-20').toDate(),
+          dueDate: '2020-10-20',
+          lineage: [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village' ],
+          overdue: true,
+          owner: 'b',
+          forId: 'b',
+        },
+      ];
+      userContactService.get.resolves(bettysContactDoc);
+      sessionService.isOnlineOnly.returns(true);
+      rulesEngineService.fetchTaskDocsForAllContacts.resolves(taskDocs);
+      lineageModelGeneratorService.reportSubjects.resolves(taskLineages);
+
+      await new Promise(resolve => {
+        sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+        getComponent();
+      });
+
+      expect(await component.currentLevel).to.be.undefined;
+      expect((<any>TasksActions.prototype.setTasksList).args).to.deep.equal([[expectedTasks]]);
+    });
+
+    it('should not change the tasks lineage if user is offline with unrelated lineage', async () => {
+      const expectedTasks = [
+        {
+          _id: 'e1',
+          date: moment('2020-10-20').toDate(),
+          dueDate: '2020-10-20',
+          lineage: [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area' ],
+          overdue: true,
+          owner: 'a',
+          forId: 'a',
+        },
+        {
+          _id: 'e2',
+          date: moment('2020-10-20').toDate(),
+          dueDate: '2020-10-20',
+          lineage: [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village' ],
+          overdue: true,
+          owner: 'b',
+          forId: 'b',
+        },
+      ];
+      userContactService.get.resolves(userContactDoc);
+      sessionService.isOnlineOnly.returns(false);
+      rulesEngineService.fetchTaskDocsForAllContacts.resolves(taskDocs);
+      lineageModelGeneratorService.reportSubjects.resolves(taskLineages);
+
+      await new Promise(resolve => {
+        sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+        getComponent();
+      });
+
+      expect(await component.currentLevel).to.equal('parent');
+      expect((<any>TasksActions.prototype.setTasksList).args).to.deep.equal([[expectedTasks]]);
+    });
+
+    it('should update the tasks lineage if user is offline with related place to lineage', async () => {
+      const expectedTasks = [
+        {
+          _id: 'e1',
+          date: new Date('2020-10-20'),
```

Comment:
Apply @jkuester about dates in this test too. 
If you still have the TZ issue, take a look at how we use `sinon.useFakeTimers`